dev/com.ibm.ws.ejbcontainer/src/com/ibm/ws/ejbcontainer/osgi/internal/EJBMDOrchestratorImpl.java

```diff
@@ -1342,32 +1344,43 @@ protected void processZOSMetadata(BeanMetaData bmd) {
         if (isTraceOn && tc.isEntryEnabled())
             Tr.entry(tc, "processZOSMetadata : " + bmd);
 
-        if (EJSPlatformHelper.isZOS()) {
-            bmd.m_syncToOSThreadValue = getSyncToOSThreadSetting(ContainerConfigConstants.syncToOSThreadSetting, bmd);
+        if (isZOS()) {
+            // Call the BeanMetaData's initializeSyncToOSThread method to check environment entry
+            bmd.initializeSyncToOSThread();
+            
+            // If not set by environment entry, check the binding
+            if (!bmd.m_syncToOSThreadValue) {
+                bmd.m_syncToOSThreadValue = getSyncToOSThreadSetting(ContainerConfigConstants.syncToOSThreadSetting, bmd);
+            }
+            
             if (bmd.m_syncToOSThreadValue) {
                 if (isTraceOn && tc.isDebugEnabled())
-                    Tr.debug(tc, "From the bean's env-var, m_syncToOSThreadValue is set to true");
-
-                // check to if server is enabled for SyncToOSThread
-                // WSLoginLocalOSExtension localSecurityLoginExt;
-                // localSecurityLoginExt = com.ibm.ws.security.auth.j2c.WSLoginLocalOSExtensionFactory.getInstance();
-                // if (!(localSecurityLoginExt.isApplicationSyncToOSThreadEnabled())) {
-                //     Tr.warning(tcWS390, "BBOJ0081", bmd.j2eeName);
-                //     bmd.m_syncToOSThreadValue = false;
-                // } else {
-                //     if (isTraceOn && tc.isDebugEnabled())
-                //         Tr.debug(tc, "EJB requests SynToOSThread, and the server is enabled for SyncToOSThread");
-                // }
+                    Tr.debug(tc, "SyncToOSThread is enabled for bean: " + bmd.enterpriseBeanName);
+
+                // Check if ThreadIdentityManager supports application thread identity
+                if (ThreadIdentityManager.isAppThreadIdentityEnabled()) {
+                    if (isTraceOn && tc.isDebugEnabled())
+                        Tr.debug(tc, "EJB requests SyncToOSThread, and the server is enabled for SyncToOSThread");
+                } else {
+                    if (isTraceOn && tc.isDebugEnabled())
+                        Tr.debug(tc, "EJB requests SyncToOSThread, but the server does not have ThreadIdentityService enabled");
+                    // Keep the setting enabled - the ThreadIdentityManager will handle the case when no service is available
+                }
             } else {
                 if (isTraceOn && tc.isDebugEnabled())
-                    Tr.debug(tc, "From the bean's env-var, m_syncToOSThreadValue is set to false as a default");
+                    Tr.debug(tc, "SyncToOSThread is not enabled for bean: " + bmd.enterpriseBeanName);
             }
         }
 
         if (isTraceOn && tc.isEntryEnabled())
             Tr.exit(tc, "processZOSMetadata");
     }
 
+    private static boolean isZOS(){
+        String osName = System.getProperty("os.name");
```

Comment:
Note : if you cannot use `EJSPlatformHelper.isZOS()`, then this methods will need to be re-written to either add a doPriv, or move the call to `System.getProperty("os.name")` to `ContainerProperties.java`.  This issue is that sometimes EJBs are initialized from an application thread, which may not have permission to obtain system properties.  Since adding doPriv blocks to everywhere EJB Container needs to access properties can add inefficiencies, instead EJB Container reads all properties it supports in one place... ContainerProperties, which is guaranteed to have permission to read system properties.  It would be much better to use EJSPlatformHelper.isZOS() and avoid this.